scripts/generate-sitemap.js

```diff
@@ -13,63 +13,99 @@ async function readSchemaJson(filePath) {
+    return null;
+  }
+  const pageName = name.split(".")[0];
+  const permalink = _path.join(relativePath, pageName);
+  const siteMapEntry = {
+    permalink,
+    title:
+      schemaData.page.title ||
+      pageName.charAt(0).toUpperCase() + pageName.slice(1),
+  };
+  // check if file is actually an index page for a directory
+  try {
+    const directoryPath = _path.join(fullPath, name.split(".")[0]);
+    const stats = await fs.stat(directoryPath);
+    if (stats.isDirectory()) {
+      return {
+        ...siteMapEntry,
+        children: await processDirectory(directoryPath, permalink),
+      };
     }
+    return siteMapEntry;
+  } catch (error) {
+    return siteMapEntry;
+  }
+}
 
-    if (entry.isDirectory()) {
-      const subRelativePath = path.join(relativePath, entry.name);
-
-      // Check if index.json exists
-      const indexFilePath = path.join(entryPath, "index.json");
-      const indexSchemaData = await readSchemaJson(indexFilePath);
+async function processDirectory(fullPath, relativePath) {
+  const entries = await fs.readdir(fullPath, { withFileTypes: true });
+  let children = [];
 
-      if (indexSchemaData) {
-        const permalink = `/${subRelativePath}`;
-        paths.push({
-          permalink,
-          title: indexSchemaData.page.title || entry.name,
-          paths: await processDirectory(entryPath, subRelativePath),
-        });
-      } else {
-        paths.push({
-          paths: await processDirectory(entryPath, subRelativePath),
-        });
+  // Check if _pages.json exists
+  const pageOrderFilePath = _path.join(fullPath, "_pages.json");
+  const pageOrderData = await readSchemaJson(pageOrderFilePath);
+  if (pageOrderData) {
+    const childPages = pageOrderData["pages"];
+    for (const child of childPages) {
+      const childEntry = await processFile(
+        fullPath,
+        relativePath,
+        child + ".json"
+      );
+      if (childEntry) {
+        children.push(childEntry);
       }
     }
+  } else {
+    // If _pages.json does not exist, process files in the directory in arbitrary order
+    const fileEntries = entries.filter((entry) => entry.isFile());
 
-    const schemaData = await readSchemaJson(entryPath);
-    if (schemaData) {
-      const pageName = entry.name.split(".").slice(0, -1).join(".");
-      const permalink = `/${path.join(relativePath, pageName)}`;
-      paths.push({
-        permalink,
-        title:
-          schemaData.title ||
-          pageName.charAt(0).toUpperCase() + pageName.slice(1),
-      });
+    for (const fileEntry of fileEntries) {
+      const childEntry = await processFile(
+        fileEntry.path,
+        relativePath,
+        fileEntry.name
+      );
+      if (childEntry) {
+        children.push(childEntry);
+      }
     }
   }
 
-  // Sort the paths array if needed, for example, by title or permalink
-  // paths.sort((a, b) => a.title.localeCompare(b.title));
+  return children;
+}
+
+async function processSchemas() {
```

Comment:
nit: I think we should name this as `processRootDirectory()` or something? I had a false impression initially that this was to process all files in the `schemas` folder recursively.